contracts/common/ScaleReputation.sol

```diff
@@ -20,35 +20,26 @@ import "../../lib/dappsys/math.sol";
 
 contract ScaleReputation is DSMath {
   // Note that scaleFactor should be a WAD.
-  function scaleReputation(int256 reputationAmount, uint256 scaleFactor) internal pure returns (int256) {
-    if (reputationAmount == 0 || scaleFactor == 0) {
-      return 0;
-    }
-
-    int256 scaledReputation;
-    int256 absAmount;
-
-    if (reputationAmount == type(int256).min){
-      absAmount = type(int256).max; // Off by one, but best we can do - probably gets capped anyway
-    } else {
-      absAmount = reputationAmount >= 0 ? reputationAmount : -reputationAmount;
-    }
-
-    int256 sgnAmount = reputationAmount >= 0 ? int(1) : -1;
+  function scaleReputation(int256 reputationAmount, uint256 scaleFactor)
+    internal
+    pure
+    returns (int256 scaledReputation)
+  {
+    if (reputationAmount == 0 || scaleFactor == 0) { return 0; }
+
+    int256 sgnAmount = (reputationAmount >= 0) ? int256(1) : -1;
+    int256 absAmount = (reputationAmount == type(int256).min)
+      ? type(int256).max // Off by one, but best we can do - probably gets capped anyway
+      : (reputationAmount >= 0) ? reputationAmount : -reputationAmount;
```

Comment:
I'm sorry, but _nested_ ternary statements is where I draw the line. Kudos for discovering a bug in that `solidity-coverage` can't lint this correctly, but the fact that bug seemingly has gone unnoticed until now is probably meaningful in of itself.

Reply:
Alright, I'll yield on this one, despite how incredible my refactor looked.

Reply:
Again, is 'looked' code for 'fewest characters'? :stuck_out_tongue:

Reply:
Ultimately I think we're jointly optimizing two variables: easy of interpreting a block of code, and ease of interpreting a function. Having concise bits of code makes it easier to parse a function or contract, even if a single section of code takes a bit longer to parse (arguable). In my mind, conditionals suggest branching behavior, so using them solely to implement require logic makes the function as a whole more difficult to parse. Ultimately, this is simple logic for setting a single variable, so I think that a condensed version leads to a cleaner overall contract, even if the statement itself requires a few more seconds to parse.